Backend framework integrated with Express and Apollo

donjewey opened the pull request:
Distinguished commits are lacking due to separate repo migration.
GraphQL and REST are both implemented in Typescript.

donjewey commented:
#5 Linked to issue #5

evanmcneely reviewed (commented):
in server/controllers/UserController.ts and server/controllers/UserController.ts...

1. It should be possible to make the properites create, read, update and delete optional (with a ? at the end of the property name) so that when extending to create the controller you don't get an error if you don't want to implement one of those methods. Currently they are all mandatory.
2. If this is express, shouldn't there be a req, res and next paramater?

shalarewicz reviewed (commented):
Once Evan's comments are addressed can you submit evidence that you're able to successfully send a request and receive a response?

donjewey commented:
Deleted the abstract CRUDController class to remove the lack of REST options in child Controllers.
Updated the GraphQL framework to include Mutations and a few more Queries.
Tested REST and GQL endpoints, REST screenshots included for evidence. 
GQL endpoints are testable on `localhost:3000/gql` , or I will gladly screen share over Zoom.

<img width="641" alt="Screen Shot 2022-05-17 at 6 06 53 PM" src="https://user-images.githubusercontent.com/95050440/168936870-a84febc4-1e85-4cd8-b2cc-8f66f463a577.png">

<img width="642" alt="Screen Shot 2022-05-17 at 6 07 43 PM" src="https://user-images.githubusercontent.com/95050440/168936874-236f0b9f-80bc-42b3-b4cd-edd728a3e444.png">

<img width="645" alt="Screen Shot 2022-05-17 at 6 07 54 PM" src="https://user-images.githubusercontent.com/95050440/168936876-0ec8862c-68aa-4e6e-aecd-3de324852f54.png">

<img width="701" alt="Screen Shot 2022-05-17 at 6 05 15 PM" src="https://user-images.githubusercontent.com/95050440/168936868-7126417e-3b4e-4292-8b63-433e8b5382e7.png">

<img width="638" alt="Screen Shot 2022-05-17 at 6 05 05 PM" src="https://user-images.githubusercontent.com/95050440/168936864-cdc862ed-26de-4d0d-bdd7-29002320d17a.png">
